go.sum

```diff
@@ -186,12 +323,15 @@ github.com/ipfs/go-ipfs-delay v0.0.0-20181109222059-70721b86a9a8/go.mod h1:8SP1Y
 github.com/ipfs/go-ipfs-delay v0.0.1 h1:r/UXYyRcddO6thwOnhiznIAiSvxMECGgtv35Xs1IeRQ=
 github.com/ipfs/go-ipfs-delay v0.0.1/go.mod h1:8SP1YXK1M1kXuc4KJZINY3TQQ03J2rwBG9QfXmbRPrw=
 github.com/ipfs/go-ipfs-ds-help v0.0.1/go.mod h1:gtP9xRaZXqIQRh1HRpp595KbBEdgqWFxefeVKOV8sxo=
-github.com/ipfs/go-ipfs-ds-help v0.1.1 h1:IW/bXGeaAZV2VH0Kuok+Ohva/zHkHmeLFBxC1k7mNPc=
 github.com/ipfs/go-ipfs-ds-help v0.1.1/go.mod h1:SbBafGJuGsPI/QL3j9Fc5YPLeAu+SzOkI0gFwAg+mOs=
-github.com/ipfs/go-ipfs-exchange-interface v0.0.1 h1:LJXIo9W7CAmugqI+uofioIpRb6rY30GUu7G6LUfpMvM=
+github.com/ipfs/go-ipfs-ds-help v1.0.0 h1:bEQ8hMGs80h0sR8O4tfDgV6B01aaF9qeTrujrTLYV3g=
+github.com/ipfs/go-ipfs-ds-help v1.0.0/go.mod h1:ujAbkeIgkKAWtxxNkoZHWLCyk5JpPoKnGyCcsoF6ueE=
```

Comment:
Please do not merge this PR, or any related to contexts in the datastore, until there are no references to go-ipfs-ds-help >v1 in the go.sum file.

Reply:
For some context (pun intended 😄) here. What's going on is that we're in the middle of adding contexts to go-datastore (in go-datastore v0.5.0) and bubbling up through the various dependencies https://github.com/ipfs/go-ipfs/issues/6803.

There's an ongoing technical debt/stress in that the ecosystem (largely go-ipfs vs newer things) in that there are two versions v0 and v1 of repos like go-ipfs-ds-help, go-ipfs-blockstore, and go-filestore where v0 stores blocks keyed on CIDs and v1 stores them keyed on multihashes.

This means that anything that uses a blockstore and is used by go-ipfs should be dependent on the v0 ones of the above and not v1, as v1 will break go-ipfs, but v0 will silently upgrade to v1 and lotus, etc. will be fine.

go-ipfs is planning to finally get this migration done in v0.12.0 as the only change in that release https://github.com/ipfs/go-ipfs/issues/6815, at which point we shouldn't have to deal with this anymore.

Reply:
hi @whyrusleeping @aschmahmann ,

Graphsync seems caught in the middle here. I'm wondering: if go-ipfs is updating in the next major release, and the high high odds are that you're not going to update go-graphsync in go-ipfs until then, what is the problem with going to the v1's now?

Reply:
also: we can always cut point releases for go-ipfs if that's super neccesary. But as go-ipfs is hardly a frequent consumer of go-graphsync -- it's still experimental as far as I know -- do we need to hold this up on go-ipfs?

Reply:
@aschmahmann Ok, upon further review, I think I understand. There are two endeavors:
1. https://github.com/ipfs/go-ipfs/issues/6803 -- adding contexts
2. Updating IPFS datastores to use multihashes instead of CIDs (v0 -> v1 versions of everything)

So I guess then we need to find the v0 versions of these repos that support the new interfaces. It's not entirely clear from https://github.com/ipfs/go-ipfs/issues/6803 what those are, especially since there's a comment from just two days ago saying there's a need to back out cause the v0 go-ipfs-blockstore wasn't done correctly.

If you can give us the exact v0 versions we should use, I can update.

Reply:
if not though, and this is urgent for Estuary, I am inclined to prioritize the needs of heavy users of go-graphsync over go-ipfs, which can continue to just use old versions until it's ready to update. (again, graphsync is to my knowledge experimental in go-ipfs, so I can't imagine keeping it super up to date is very important -- maybe I'm missing something though?)

Reply:
@hannahhoward the complexity here isn't really about the v0/v1 blockstore, but more about the go-datastore v0.5.0 breaking changes.

This PR is now up to speed, however we're left with https://github.com/ipfs/go-graphsync/pull/275#discussion_r753475830 and can continue in that thread